### PR TITLE
Test case for getFileName method

### PR DESCRIPTION
**Pull request by prabushi**
This test methods checks whether the file name is correctly extracted from the file path.

### CODE REVIEW

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 1]
```diff
@@ -170,4 +168,29 @@ public void testFileReadChars() {
...
+					jaggeryServerConnection.getInputStream()));
+
+			String inputLine;
+			while ((inputLine = in.readLine()) != null) {
+				finalOutput = inputLine;
+			}
+
+			in.close();
+		} catch (IOException e) {
+			e.printStackTrace();
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Please use logger for printing the error.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 2]
```diff
@@ -170,4 +168,29 @@ public void testFileReadChars() {
...
+			URLConnection jaggeryServerConnection = jaggeryURL.openConnection();
+			BufferedReader in = new BufferedReader(new InputStreamReader(
+					jaggeryServerConnection.getInputStream()));
+
+			String inputLine;
+			while ((inputLine = in.readLine()) != null) {
+				finalOutput = inputLine;
+			}
+
+			in.close();
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Close the input stream in the finally block.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 3]
```diff
@@ -170,4 +168,29 @@ public void testFileReadChars() {
           
       }
 
+	@Test(groups = {"jaggery"},
+			description = "Test for get file name operation")
+	public void testGetFileName() {
+		ClientConnectionUtil.waitForPort(9763);
+
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java:90]: Remove the unnecessary blank lines.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java (hunk 1 above)
Comment (prabushi) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Committed the changes.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java (hunk 2 above)
Comment (prabushi) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Changed the code.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java (hunk 3 above)
Comment (prabushi) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java:90]: Removed the blanks.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 4]
```diff
@@ -35,139 +35,163 @@
...
         			jaggeryServerConnection.getInputStream()));
-        
+
           	String inputLine;
   			while ((inputLine = in.readLine()) != null) {
   				finalOutput = inputLine;
   			}
-			    
+
 			in.close();
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Input stream needs to be closed in finally block.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 5]
```diff
@@ -35,139 +35,163 @@
...
 		} catch (IOException e) {
 			e.printStackTrace();
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Use logger to log error.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 6]
```diff
@@ -35,139 +36,138 @@
...
+        BufferedReader in = null;
+        try {
+            URL jaggeryURL = new URL("http://localhost:9763/testapp/file.jag");
+            URLConnection jaggeryServerConnection = jaggeryURL.openConnection();
+            in = new BufferedReader(new InputStreamReader(jaggeryServerConnection.getInputStream()));
+            String inputLine;
+            while ((inputLine = in.readLine()) != null) {
+                finalOutput = inputLine;
+            }
+            in.close();
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Remove this in.close() as it's already being closed in finally.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 7]
```diff
@@ -35,139 +36,138 @@
...
+        } catch (IOException e) {
+            log.error(e);
+        } finally {
+            in.close();
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: It's a good idea to to check in == null before trying to do any operation. Please change that.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 8]
```diff
@@ -35,139 +36,138 @@
...
+            URL jaggeryURL = new URL("http://localhost:9763/testapp/file.jag");
+            URLConnection jaggeryServerConnection = jaggeryURL.openConnection();
+            in = new BufferedReader(new InputStreamReader(jaggeryServerConnection.getInputStream()));
+            String inputLine;
+            while ((inputLine = in.readLine()) != null) {
+                finalOutput = inputLine;
+            }
+            in.close();
+        } catch (IOException e) {
+            log.error(e);
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Add some application specific information when you are logging the error.

File: modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java [hunk 9]
```diff
@@ -35,139 +36,144 @@
...
+            String inputLine;
+            if(in != null) {
+                while ((inputLine = in.readLine()) != null) {
+                    finalOutput = inputLine;
+                }
+            }
+        } catch (IOException e) {
+            log.error("Reading InputStream fails in fileExist test method", e);
+        } finally {
+            in.close();
```
Comment (sinthuja) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: If there is a error while opening the file, then there will be an exception. In that case inputstream - 'in' will be null, and hence this statement will cause NPE. Please do a null check before you close the stream.

Comment (prabushi) [inline modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java]: Added null check before inputstream close.

**Pull request merged**